docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,236 @@
+
+While an "externally managed" InfraCluster won't reconcile or manage the lifecycle of the cluster infrastructure, CAPI will still be able to create compute nodes within it.
+
+The machine controller must be able to operate without hard dependencies regardless of the cluster infrastructure being managed or externally managed.
+![](https://i.imgur.com/nA61XJt.png)
+
+### User Stories
+
+#### Story 1 - Alternate control plane provisioning with user managed infrastructure
+As a cluster provider I want to use CAPI in my service offering to orchestrate Kubernetes bootstraping while letting workload cluster operators own their infrastructure lifecycle.
+
+#### Story 2 - Restricted access to cloud provider APIs
+As a cluster operator I want to use CAPI to orchestrate kubernetes bootstraping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.
+
+#### Story 3 - Consuming existing cloud infrastructure
+As a cluster operator I want to use CAPI to orchestate Kubernetes bootstraping while reusing infrastructure that has already been created in the organisation either by me or another team.
+
+### Implementation Details/Notes/Constraints
+
+**Managed**
+
+- It will be default and will preserve existing behaviour. An InfraCluster CR without the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation.
+
+
+**Externally Managed**
+
+An InfraCluster CR with the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation.
+
+The provider InfraCluster controller must:
+- Skip any reconciliation of the resource.
+
+- Not update the resource or its status in any way
+
+The external management system must:
+
+- Populate all required fields within the InfraCluster spec to allow other CAPI components to continue as normal.
+
+- Adhere to all Cluster API contracts for infrastructure providers.
+
+- When the infrastructure is ready, set the appropriate status as is done by the provider controller today.
+
+#### Provider implementation changes
+
+To enable providers to implement the changes required by this contract, Cluster API is going to provide a new `predicates.ResourceExternallyManaged` predicate as part of its utils.
+
+This predicate filters out any resource that has been marked as "externally managed" and prevents the controller from reconciling the resource.
+
+### Security Model
+
+When externally managed, the required cloud provider privileges are significantly reduced when compared with a traditionally managed cluster.
+The only privileges required by CAPI are those that are required to manage machines.
+
+For example, when an AWS cluster is managed by CAPI, persmissions are required to be able to create VPCs and other networking components that are managed by the AWSCluster controller. When externally managed, these permissions are not required as the external entity is reponsible for creating such components.
```

Comment:
Okay reading this it makes a little more sense. I think this is only true for AWS. Other clouds like Azure don't operate with per resource permissions, instead you would give your service principal reader/contributor/owner permissions at a certain scope, like the resource group for example.

In that case, I would change the above sentence to "The cloud provider privileges required by CAPI might be reduced when compared with a managed cluster infrastructure."